### PR TITLE
add YAML config for imagestreams generator

### PR DESCRIPTION
**Pull request by zmiklank**
Depends on: https://github.com/sclorg/ci-scripts/pull/119

### CODE REVIEW

File: imagestreams/imagestreams.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  pretty_name: PostgreSQL
+  sample_repo: ""
+  category: database
+  description: >
+    Provides a PostgreSQL APP_VERSION database on DISTRO_NAME.
+    For more information about using this database image,
+    including OpenShift considerations,
+    see https://github.com/sclorg/postgresql-container/blob/master/README.md.
+  imagestream_files:
+  - filename: postgresql-centos_stream.json
```
Comment (phracek) [inline imagestreams/imagestreams.yaml:12]:
```suggestion
  - filename: postgresql-centos.json
```

File: imagestreams/postgresql-rhel.json [hunk 2]
```diff
@@ -10,217 +10,203 @@
...
         },
         "referencePolicy": {
           "type": "Local"
         }
       },
       {
-        "name": "12",
+        "name": "15-el9",
         "annotations": {
-          "openshift.io/display-name": "PostgreSQL (Ephemeral) 12",
```
Comment (phracek) [inline imagestreams/postgresql-rhel.json:155]: I guess, the name should be the same `PostgreSQL (Ephemeral) 12` and not `PostgreSQL 12 (Ephemeral)`.

**Review by phracek — CHANGES_REQUESTED** [review]
It would be good to also add GitHub Action that will check if imagestreams were properly generated or not like dist-gen use case. The sort function in the generator would help.

File: imagestreams/postgresql-rhel.json (hunk 2 above)
Comment (zmiklank) [reply imagestreams/postgresql-rhel.json:155]: This seems to me like the original version being incorrect, as all other display names have same format, except for the Ephemeral one.

File: imagestreams/postgresql-rhel.json [hunk 3]
```diff
@@ -10,217 +10,203 @@
...
-        "name": "10",
+        "name": "latest",
         "annotations": {
-          "openshift.io/display-name": "PostgreSQL (Ephemeral) 10",
+          "openshift.io/display-name": "PostgreSQL 15 (Latest)",
           "openshift.io/provider-display-name": "Red Hat, Inc.",
-          "description": "Provides a PostgreSQL 10 database on RHEL 7. For more information about using this database image, including OpenShift considerations, see https://github.com/sclorg/postgresql-container/blob/master/README.md.",
+          "description": "Provides a PostgreSQL 15 database on RHEL 8. For more information about using this database image, including OpenShift considerations, see https://github.com/sclorg/postgresql-container/blob/master/README.md.\n",
           "iconClass": "icon-postgresql",
-          "tags": "database,postgresql,hidden",
```
Comment (zmiklank) [inline imagestreams/postgresql-rhel.json:213]: Also this one tag is different in the Ephemeral streams.

Comment (zmiklank) [reply imagestreams/postgresql-rhel.json:213]: Discussed this with team, concluded that ephemeral streams are no longer needed and will be removed.

File: imagestreams/postgresql-rhel.json (hunk 2 above)
Comment (zmiklank) [reply imagestreams/postgresql-rhel.json:155]: Ephemeral streams are now removed.

**Comment by zmiklank** [comment]
> It would be good to also add GitHub Action that will check if imagestreams were properly generated or not like dist-gen use case.

I totally agree. I can create some github action for it.

> The sort function in the generator would help.

Could you please explain further?

**Review by phracek — APPROVED** [review]

**Comment by zmiklank** [comment]
[test-openshift]

**Comment by phracek** [comment]
This is blocked by https://github.com/sclorg/postgresql-container/pull/553 which fixes OpenShift 4 errors.

**Comment by zmiklank** [comment]
[test-openshift]

**Comment by zmiklank** [comment]
Rebased. [test-openshift]

**Comment by phracek** [comment]
[test-openshift]

**Comment by zmiklank** [comment]
[test-openshift]

File: imagestreams/postgresql-rhel.json [hunk 4]
```diff
@@ -10,104 +10,90 @@
...
       },
       {
-        "name": "15-el8",
+        "name": "10-el8",
         "annotations": {
-          "openshift.io/display-name": "PostgreSQL 15 (RHEL 8)",
+          "openshift.io/display-name": "PostgreSQL 10 (RHEL 8)",
           "openshift.io/provider-display-name": "Red Hat, Inc.",
-          "description": "Provides a PostgreSQL 15 database on RHEL 8. For more information about using this database image, including OpenShift considerations, see https://github.com/sclorg/postgresql-container/blob/master/README.md.",
+          "description": "Provides a PostgreSQL 10 database on RHEL 8. For more information about using this database image, including OpenShift considerations, see https://github.com/sclorg/postgresql-container/blob/master/README.md.\n",
```
Comment (phracek) [inline imagestreams/postgresql-rhel.json:71]: The new line at the end is not needed.

**Review by phracek — CHANGES_REQUESTED** [review]
On more issue with the new line at the end of description

File: imagestreams/postgresql-rhel.json (hunk 4 above)
Comment (zmiklank) [reply imagestreams/postgresql-rhel.json]: Good catch. That is some funny behaviour of yaml. I've fixed it with https://github.com/sclorg/postgresql-container/pull/549/commits/196423e6d8a834229b07d5e9c737104d925c5e70.

**Comment by zmiklank** [comment]
Depends on: https://github.com/sclorg/ci-scripts/pull/121

**Comment by phracek** [comment]
[test-openshift]

**Pull request merged**